tests/dialects/test_memref.py

```diff
@@ -140,6 +142,32 @@ def test_memref_alloc():
         alloc6.verify()
 
 
+def test_memref_global():
+    memref_type = memref.MemRefType(i32, (2, 2))
+    tensor_type = builtin.TensorType(i32, (2, 2))
+    alignment = 65
+    global1 = memref.Global.get(
+        StringAttr("my_global"),
+        memref_type,
+        DenseIntOrFPElementsAttr.from_list(tensor_type, [1, 2, 3, 4]),
+        sym_visibility=StringAttr("public"),
+        alignment=builtin.IntegerAttr(alignment, 64),
+    )
+    global2 = memref.Global.get(
+        StringAttr("my_valid_global"),
+        memref_type,
+        DenseIntOrFPElementsAttr.from_list(tensor_type, [1, 2, 3, 4]),
+        sym_visibility=StringAttr("public"),
+        alignment=64,
+    )
+    with pytest.raises(
+        VerifyException,
+        match=f"Alignment attribute {alignment} is not a power of 2",
+    ):
+        global1.verify()
+    global2.verify()
```

Comment:
Can you please make the failing version of this a failing filecheck test instead? And I think that the memref_ops filecheck already convers the non-failing case, so we should be ok

Reply:
Sure, why a filecheck instead of a pytest test?

Reply:
It's just the thing we've been going for in the project. I usually prefer pytest tests but it's probably worth doing for consistency. It also couples the test less with the Python API.